ext/nio4r/bytebuffer.c

```diff
@@ -75,6 +75,7 @@ void Init_NIO_ByteBuffer()
 static VALUE NIO_ByteBuffer_allocate(VALUE klass)
 {
     struct NIO_ByteBuffer *bytebuffer = (struct NIO_ByteBuffer *)xmalloc(sizeof(struct NIO_ByteBuffer));
+    bytebuffer->buffer = NULL;
```

Comment:
Thanks for doing this! This code was from a GSoC project, but it seems I missed this while reviewing it and it really should be properly initialized at the time it's allocated.

Another approach that ensures the buffer is properly allocated would be to malloc a 0 byte buffer, then realloc it once initialize is called. This seems like the safest option to me, although not necessarily the most efficient.